powershell/upload/UploadFile.ps1

```diff
@@ -0,0 +1,533 @@
+        Write-Host "Credentials saved."
+    }
+
+    [Credentials] Retrieve() {
+        Write-Host "----"
+        Write-Host "Retrieving your credentials."
+        Write-Host "| Path: $($this._credentialsPath)"
+
+        if (-not (Test-Path -Path $this._credentialsPath -PathType Leaf)) {
+            Write-Host "----"
+            Write-Host "Credentials not found."
+            Write-Host "| Path: $($this._credentialsPath)"
+            
+            return $null
+        }
+
+        $credentialsStorage = Import-CliXml -Path $this._credentialsPath
+
+        $credentials = [Credentials]::new()
+        $credentials.ClientId = $credentialsStorage.GetNetworkCredential().UserName
+        $credentials.ClientSecret = $credentialsStorage.GetNetworkCredential().Password
+
+        Write-Host "Credentials retrieved."
+
+        return $credentials
+    }
+}
+
+class FileApiService {
+    hidden [FileApiClient] $_fileApiClient
+    hidden [string] $_tenantId
+    hidden [long] $_uploadSizeLimit
+    hidden [string] $_boundary
+    hidden [string] $_businessTypeId
+
+    FileApiService(
+        [FileApiClient] $fileApiClient,
+        [string] $tenantId,
+        [string] $businessTypeId
+    ) {
+        $this._fileApiClient = $fileApiClient
+        $this._tenantId = $tenantId
+        $this._boundary = "file_info"
+        $this._businessTypeId = $businessTypeId
+
+        # API supports files up to 100 megabytes
+        $this._uploadSizeLimit = 100 * 1024 * 1024
+    }
+
+    [string] CreateFileToUpload([string] $contentFilePath) {
+        $headerFilePath = ""
+        $footerFilePath = ""
+        try {
+            $folderPath = $(Split-Path -Path $contentFilePath)
+            $contentFilename = $(Split-Path -Path $contentFilePath -Leaf)
+            $createdFilePath = "$($folderPath)\$([Helper]::ConverToUniqueFileName("multipart.bin"))"
+
+            $headerFilename = [Helper]::ConverToUniqueFileName("header.txt")
+            $headerFilePath = "$($folderPath)\$($headerFilename)"
+            $headerContent = "--$($this._boundary)`r`n" # Windows line breaks are required.
+            $headerContent += "Content-Type: application/json; charset=UTF-8`r`n"
+            $headerContent += "`r`n"
+            $headerContent += "{`r`n`"name`":`"$($contentFilename)`",`r`n`"businesstypeid`":`"$($this._businessTypeId)`"`r`n}`r`n"
+            $headerContent += "--$($this._boundary)`r`n`r`n"
+
+            $footerFilename = [Helper]::ConverToUniqueFileName("footer.txt")
+            $footerFilePath = "$($folderPath)\$($footerFilename)"
+            $footerContent = "`r`n--$($this._boundary)--"
+
+            New-Item -Path $folderPath -Name $headerFilename -Value $headerContent
+            New-Item -Path $folderPath -Name $footerFilename -Value $footerContent
+
+            cmd /c copy /b $headerFilePath + $contentFilePath + $footerFilePath $createdFilePath
+            return $createdFilePath
+        }
+        finally {
```

Comment:
If the code explodes before creating the files, it will explodes again when trying to delete them because they don't exist.
Maybe you could move the try part bellow, just before the creation of the files.
If you do it like that, you could remove the
        $headerFilePath = ""
        $footerFilePath = ""

Reply:
With Test-Path I can check if the file exists and only in this case call to Remove-Item